src/rabbitmq/RabbitmqClient.ts

```diff
@@ -71,49 +76,63 @@ export abstract class RabbitmqClient {
     this.readyGate.channelNotReady();
   }
 
-  async init() {
-    await this.connect();
+  async init(addHandler = true) {
+    await this.connect(addHandler);
   }
 
   /**
    * Connect to RabbitMQ broker
    */
-  protected async connect(): Promise<void> {
-    await this.createConnection();
-    await this.createChannel();
-    this.reconnecting = false;
+  protected async connect(addHandler = true): Promise<void> {
+    await this.createConnection(addHandler);
+    await this.createChannel(addHandler);
+    this.stopReconnecting();
     this.readyGate.channelReady();
   }
 
-  private async createConnection() {
+  private async createConnection(addHandler: boolean): Promise<void> {
```

Comment:
need a default value for addHandler here just in case this being called somewhere else?